### ISSUE TITLE
Can't use class variables in virtual type

### ISSUE THREAD

**Issue by gdotdesign** [crystal-lang/crystal#916]
The following code throws an error, I think it shouldn't http://play.crystal-lang.org/#/r/653

Here is the error:
```
cast to Crystal::ClassVarContainer failed
*raise<String>:NoReturn +70 [0]
*Crystal::TypeVisitor#lookup_class_var<Crystal::TypeVisitor, Crystal::ClassVar, Bool>:Crystal::Var+ +480 [0]
*Crystal::ASTNode+@Crystal::ASTNode#accept<Crystal::ASTNode+, Crystal::TypeVisitor>:Nil +6012 [0]
*Crystal::Call#instantiate<Crystal::Call, Crystal::Matches, Crystal::Type+, Nil>:Array(Crystal::Def+) +12439 [0]
*Crystal::Call#lookup_matches_in_type<Crystal::Call, Crystal::Type+, Array(Crystal::Type+), Nil, String>:Array(Crystal::Def+) +12384 [0]
*Crystal::Call#recalculate<Crystal::Call>:Array(Crystal::Def+)? +2199 [0]
*Crystal::TypeVisitor#visit<Crystal::TypeVisitor, Crystal::Call>:Bool +4705 [0]
*Crystal::ASTNode+@Crystal::ASTNode#accept<Crystal::ASTNode+, Crystal::TypeVisitor>:Nil +2001 [0]
*Crystal::TypeVisitor#visit<Crystal::TypeVisitor, Crystal::Call>:Bool +1685 [0]
*Crystal::ASTNode+@Crystal::ASTNode#accept<Crystal::ASTNode+, Crystal::TypeVisitor>:Nil +2001 [0]
*Crystal::ASTNode+@Crystal::ASTNode#accept<Crystal::ASTNode+, Crystal::TypeVisitor>:Nil +27946 [0]
*Crystal::Compiler#compile<Crystal::Compiler, Array(Crystal::Compiler::Source), String>:Crystal::Compiler::Result +4739 [0]
*Crystal::Command::run<Array(String)>:(Nil | Bool | FileDescriptorIO+ | Crystal::Compiler::Result | Array(Crystal::Init::View+:Class)) +14807 [0]
main +17080 [0]
__libc_start_main +245 [0]
_start +41 [0]

Error: you've found a bug in the Crystal compiler. Please open an issue: https://github.com/manastech/crystal/issues
```

**Comment by asterite** [comment]
Reduced:

```crystal
class Parent
  def classvar
    @@classvar
  end
end

class Child < Parent
end

class Child2 < Parent
end

(Child.new || Child2.new).classvar
```

The problem is that the type of `Child.new || Child2.new` is `Parent+`, and we didn't think of the case of handling class variables for that type. I think in this case `@@classvar` should be associated to Parent, because class variables are not inherited, but we have to think about this a bit more (and think about the uses we want to give class variables, and their semantic).

**Comment by asterite** [comment]
Closed as a duplicate of #2458
